core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/fuse/TestFuseCrossJoinedGlobalAggregations.java

```diff
@@ -0,0 +1,503 @@
+                                Optional.empty(),
+                                SINGLE,
+                                project(
+                                        ImmutableMap.of(
+                                                "aggr_mask_right", PlanMatchPattern.expression("colRight = 2")),
+                                        tableScan(TEST_TABLE_NAME, ImmutableMap.of("colRight", "col")))));
+    }
+
+    @Test
+    public void testTableScanWithFilterOnlyOnOneTheRLeftSide()
+    {
+        tester().assertThat(new FuseCrossJoinedGlobalAggregations())
+                .on(p -> {
+                    Symbol colLeft = p.symbol("colLeft");
+                    Symbol colRight = p.symbol("colRight");
+                    Symbol countLeft = p.symbol("countLeft");
+                    Symbol sumRight = p.symbol("sumRight");
+
+                    return p.join(
+                            INNER,
+                            p.aggregation(aggregationBuilder -> aggregationBuilder
+                                    .globalGrouping()
+                                    .addAggregation(countLeft, expression("count(*)"), ImmutableList.of())
+                                    .source(p.filter(
+                                            expression("colLeft = 1"),
+                                            p.tableScan(TEST_TABLE_HANDLE, ImmutableList.of(colLeft), ImmutableMap.of(colLeft, TEST_TABLE_COLUMN))))),
+                            p.aggregation(aggregationBuilder -> aggregationBuilder
+                                    .globalGrouping()
+                                    .addAggregation(sumRight, expression("sum(colRight)"), ImmutableList.of(BIGINT))
+                                    .source(
+                                            p.tableScan(TEST_TABLE_HANDLE, ImmutableList.of(colRight), ImmutableMap.of(colRight, TEST_TABLE_COLUMN)))));
+                })
+                .matches(
+                        aggregation(
+                                globalAggregation(),
+                                ImmutableMap.of(
+                                        Optional.of("countLeft"), functionCall("count", ImmutableList.of()),
+                                        Optional.of("sumRight"), functionCall("sum", ImmutableList.of("colLeft"))),
+                                ImmutableList.of(),
+                                ImmutableList.of("aggr_mask_left"),
+                                Optional.empty(),
+                                SINGLE,
+                                project(
+                                        ImmutableMap.of(
+                                                "aggr_mask_left", PlanMatchPattern.expression("colLeft = 1")),
+                                        tableScan(TEST_TABLE_NAME, ImmutableMap.of("colLeft", "col")))));
+    }
+
+    // this fails due to test infra assumption that projection always has unique target expression
```

Comment:
Can we somehow fix this? Because it seems like a pretty imp test

Reply:
fixing the entire infra is hard and I don't know if we want to do it just for this case because it makes a lot of things easier.
Also, I have this case tested in `AbstractTestJoinQueries.testFuseCrossJoinOnGlobalAggregationWithDuplicatedAggregation`.